# Unless explicitly silenced

Errors should not pass silently, yes!

But ... sometimes you do want to explicitly silence a specific error.

The contextlib module has a context manager you can use for this, called `suppress`.

Say you have some cleanup / teardown code where you attempt to remove a file or close a connection, but if it fails, it doesn't impact the overall functionality.

For example:

```
import os
from contextlib import suppress

# ... some operations with a temporary file ...

# Cleanup
with suppress(FileNotFoundError):
    os.remove('temporary_file.tmp')
```

If temporary_file.tmp doesn't exist or has already been removed, the FileNotFoundError is ignored, allowing the program to proceed without raising an exception. This to me seems a valid use case of "explicitly silencing" this error.

However use this judiciously, to quote the suppress docs:

> "As with any other mechanism that completely suppresses exceptions, this context manager should be used only to cover very specific errors where silently continuing with program execution is known to be the right thing to do."

#zen
